File: defaultcode/cpp/singleton/singleton.cpp
#include <iostream>

using namespace std;

class $classname {
    $classname *instance;

    
    //constructor
    $classname() {
    }


    public:
        //method to get instance
        $classname *getInstance() {
            if (!instance)
                instance = new $classname;

            return instance;
        }

        //$getters

        //$setters

};

File: defaultcode/Java/singleton/singleton.java
class $classname
{
    //single instance
    private static $classname instance = null;
  
  
    // private constructor
    private $classname()
    {

    }
  
    //get instance
    public static $classname getInstance()
    {
        if (instance == null)
            instance = new $classname();
  
        return instance;
    }
}